nhrpd/nhrp_interface.c

```diff
@@ -329,7 +329,9 @@ int nhrp_interface_delete(int cmd, struct zclient *client, zebra_size_t length,
 	debugf(NHRP_DEBUG_IF, "if-delete: %s", ifp->name);
 	if_set_index(ifp, ifp->ifindex);
 	nhrp_interface_update(ifp);
-	/* if_delete(ifp); */
+
+	if_set_index(ifp, IFINDEX_INTERNAL);
```

Comment:
It seems to me that the `if_set_index(ifp, ifp->ifindex)` call a few lines above is completely redundant. Couldn't we remove it?

Reply:
yes correct, missed that fixing